src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -36,23 +36,23 @@
 #include "utils/ruleutils.h"
 #include "utils/syscache.h"
 
-/*
- * Flags that can be passed to GetForeignKeyOids to indicate
- * which foreign key constraint OIDs are to be extracted
- */
-typedef enum ExtractForeignKeyConstrainstMode
-{
-	/* extract the foreign key OIDs where the table is the referencing one */
-	INCLUDE_REFERENCING_CONSTRAINTS = 1 << 0,
 
-	/* extract the foreign key OIDs the table is the referenced one */
-	INCLUDE_REFERENCED_CONSTRAINTS = 1 << 1,
+#define BehaviorIsRestrictOrNoAction(x) \
+	((x) == FKCONSTR_ACTION_NOACTION || (x) == FKCONSTR_ACTION_RESTRICT)
+
+
+typedef bool (*CheckRelationFunc)(Oid);
 
-	/* exclude the self-referencing foreign keys */
-	EXCLUDE_SELF_REFERENCES = 1 << 2
-} ExtractForeignKeyConstraintMode;
 
 /* Local functions forward declarations */
+static void EnsureReferencingTableNotReplicated(Oid referencingTableId);
+static void EnsureSupportedFKeyOnDistKey(Form_pg_constraint constraintForm);
+static void EnsureSupportedFKeyBetweenCitusLocalAndRefTable(Form_pg_constraint
+															constraintForm,
+															char
```

Comment:
Minor: the indentation seems a bit off here, is it possible to have both the type and name in one line here?

Reply:
Unfortunately no, our indentation tool doesn't allow it